File: app/src/main/java/com/iita/akilimo/utils/enums/EnumMaizeProduceType.java
package com.iita.akilimo.utils.enums;

public enum EnumMaizeProduceType {
    GRAIN {
        @Override
        public String produce() {
            return "grain";
        }
    },
    DRY_COB {
        @Override
        public String produce() {
            return "dry_cob";
        }
    },
    FRESH_COB {
        @Override
        public String produce() {
            return "fresh_cob";
        }
    };

    public abstract String produce();
}

File: app/src/main/java/com/akilimo/mobile/utils/enums/EnumMaizeProduceType.kt
package com.akilimo.mobile.utils.enums

enum class EnumMaizeProduceType {
    GRAIN {
        override fun produce(): String {
            return "grain"
        }
    },
    DRY_COB {
        override fun produce(): String {
            return "dry_cob"
        }
    },
    FRESH_COB {
        override fun produce(): String {
            return "fresh_cob"
        }
    };

    abstract fun produce(): String
}
